src/Players/AIPlayer.hs

```diff
@@ -79,10 +79,28 @@ minimize (GameTree chessBoard score children) colour = (GameTree chessBoard scor
 
 
 -- Returns the next move that maximizes the score
--- TODO: implement + test this [Yiyi]
-getMoveWithMaxScore :: GameTree -> ChessMove
-getMoveWithMaxScore gameTree = ChessMove ('z', -1) ('z', -1)
-
+-- TODO: test this [Yiyi]
+getMoveWithMaxScore :: GameTree -> ChessPieceColour -> ChessMove
+getMoveWithMaxScore (GameTree _ _ nextMoves) colour = moveOfMaxScore
+    where movesWithScore = zip allMoves scoreOfMoves
```

Comment:
are you sure that `allMoves` and `scoreOfMoves` have stable order? As in, you're assuming that the order of `allMoves` and `scoreofMoves` is the the same as that of `nextMoves` - this might be true but it's best to test it. 

Alternatively, you can skip the zip and just do
```haskell
movesWithScore  = [(move, (getScore tree)) | (MoveSubtree move tree) <- nextMoves]
```